pom.xml

```diff
@@ -1124,6 +1124,35 @@ under the License.
 			</build>
 		</profile>
 
+		<profile>
+			<id>java21</id>
+			<activation>
+				<jdk>[21,)</jdk>
+			</activation>
+
+			<properties>
+				<!-- Bump Scala because before 2.12.18 doesn't compile on Java 21. -->
+				<scala.version>2.12.18</scala.version>
```

Comment:
Doing this will be a breaking change for Flink users that are upgrading from a lower Flink version to this Flink version with a newer Java version, since Scala 2.12.8 and higher have a breaking binary compatibility change that impacts Flink snapshots

Reply:
Unless it works the same as with Java 17 (but that I don't know)